src/patients/edit/EditPatient.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory, useParams } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { Spinner } from '@hospitalrun/components'
+
+import GeneralInformation from '../GeneralInformation'
+import useTitle from '../../page-header/useTitle'
+import Patient from '../../model/Patient'
+import { updatePatient, fetchPatient } from '../patient-slice'
+import { RootState } from '../../store'
+import { getPatientFullName, getPatientName } from '../util/patient-name-util'
+
+const getFriendlyId = (p: Patient): string => {
+  if (p) {
+    return p.friendlyId
+  }
+
+  return ''
+}
+
+const EditPatient = () => {
+  const { t } = useTranslation()
+  const history = useHistory()
+  const dispatch = useDispatch()
+
+  const [patient, setPatient] = useState({} as Patient)
+  const { patient: reduxPatient } = useSelector((state: RootState) => state.patient)
+
+  useTitle(
+    `${t('patients.editPatient')}: ${getPatientFullName(reduxPatient)} (${getFriendlyId(
+      reduxPatient,
+    )})`,
+  )
+
+  useEffect(() => {
+    setPatient(reduxPatient)
+  }, [reduxPatient])
+
+  const { id } = useParams()
+  useEffect(() => {
+    if (id) {
+      dispatch(fetchPatient(id))
+    }
+  }, [id, dispatch])
+
+  const onCancel = () => {
+    history.push(`/patients/${id}`)
+  }
+
+  const onSave = () => {
+    dispatch(
+      updatePatient(
+        {
+          ...patient,
+          fullName: getPatientName(patient.givenName, patient.familyName, patient.suffix),
+        },
+        history,
+      ),
+    )
+  }
+
+  const onFieldChange = (key: string, value: string | boolean) => {
+    setPatient({
+      ...patient,
+      [key]: value,
+    })
+  }
+
+  // see comment in ViewPatient
+  if (!reduxPatient) {
```

Comment:
`isLoading` makes sense to me, but what does the check for `!reduxPatient.id` add? or should it be `!id`, like if there's no parameter? but that seems like it would be handled separately with an error or something

Reply:
so far I've just used `isLoading` which seems to work. Now, you actually see the loading icon when you change patients, etc.